### PR TITLE
Add tests for BN API's (bn_print.c and bn_const.c)

### PR DESCRIPTION
**Pull request by slontis**
Add Tests for bn_print.c functions BN_print_fp(), BN_options(). Add Tests for bn_const.c BN_get_rfc2409_prime_768(), BN_get_rfc3526_prime_1536() & BN_get_rfc3526_prime_6144() (Note that the other getters are accessed via TLS tests).

##### Checklist

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by paulidale** [comment]
CI relevant?

**Comment by slontis** [comment]
Not sure how that built locally...

**Review by paulidale — APPROVED** [review]

File: test/bntest.c [hunk 1]
```diff
@@ -3198,6 +3198,50 @@ static int file_test_run(STANZA *s)
...
+    if (!TEST_ptr(bn = BN_new()))
+        return 0;
+    if (!TEST_ptr(bio = BIO_new_fp(stdout, BIO_NOCLOSE | BIO_FP_TEXT)))
+        goto err;
+
+    if (!TEST_true(BN_set_word(bn, 0x1234)))
+        goto err;
+    if (!TEST_true(BN_print_fp(stdout, bn)))
+        goto err;
+    BIO_printf(bio, "\n");
```
Comment (levitte) [inline test/bntest.c:3220]: Had it been me, I would rather have `bio` be a `BIO_s_mem()`, print to that using `BN_print()`, and verify the resulting string against a known string.

With the test as you currently have it, you're kind testing two things, 1) that we know how to wrap a `FILE*` into a BIO, and 2) that we have a function to print bignums...  but it's not in any way verifying that `BIO_print()` (or for that matter, `BIO_print_fp()`) actually does its job...

So, what did you actually intend to test?

File: test/bntest.c [hunk 2]
```diff
@@ -3198,6 +3198,50 @@ static int file_test_run(STANZA *s)
     return 0;
 }
 
+static int test_print(void)
+{
+    int ret = 0;
```
Comment (tom-cosgrove-arm) [inline test/bntest.c:3203]: shouldn't this be under the `#else`?

File: test/bntest.c [hunk 3]
```diff
@@ -3198,6 +3198,50 @@ static int file_test_run(STANZA *s)
...
+#endif
+    return ret;
+}
+
+static int test_dh_named_primes(void)
+{
+    int ret;
+    BIGNUM *bn1 = NULL, *bn2 = NULL, *bn3 = NULL;
+
+    ret = TEST_ptr(bn1 = BN_get_rfc2409_prime_768(NULL))
```
Comment (tom-cosgrove-arm) [inline test/bntest.c:3236]: why not just declare at this point?

```
int ret = TEST_ptr(bn1 = BN_get_rfc2409_prime_768(NULL))
```

**Comment by tom-cosgrove-arm** [comment]
Pig @slontis there are a few comments on this PR

File: test/bntest.c (hunk 1 above)
Comment (slontis) [reply test/bntest.c:3220]: Its a coverage test for BN_print_fp().

**Pull request closed without merging**